flink-architecture-tests/flink-architecture-tests-production/src/main/java/org/apache/flink/architecture/rules/ConnectorRules.java

```diff
@@ -41,16 +39,14 @@ public class ConnectorRules {
         "org.apache.flink.connector..", "org.apache.flink.streaming.connectors.."
     };
 
-    private static DescribedPredicate<JavaClass> areNotPublicAndResideOutsideOfPackages(
-            String... packageIdentifiers) {
-        return JavaClass.Predicates.resideOutsideOfPackages(packageIdentifiers)
-                .and(
-                        not(areDirectlyAnnotatedWithAtLeastOneOf(
-                                        Public.class, PublicEvolving.class))
-                                .and(not(modifier(PUBLIC))))
+    private static DescribedPredicate<JavaClass>
+            areFlinkClassesThatResideOutsideOfConnectorPackagesAndAreNotPublic() {
+        return JavaClass.Predicates.resideInAPackage("org.apache.flink..")
+                .and(JavaClass.Predicates.resideOutsideOfPackages(CONNECTOR_PACKAGES))
+                .and(not(areDirectlyAnnotatedWithAtLeastOneOf(Public.class, PublicEvolving.class)))
```

Comment:
We should _somehow_ replicate the logic of also checking the outer class for annotations; right now this creates a lot of false-positives.

Similar to what we do for method checks:
```
                            .areDeclaredInClassesThat(
                                    areJavaClasses()
                                            .and(
                                                    areDirectlyAnnotatedWithAtLeastOneOf(
                                                            PublicEvolving.class)))
```

Not sure yet how to express this properly in archunit.

Reply:
makes sense, coding that

Reply:
thanks, but I got a simpler solution

Reply:
yes belongTo should work, at least according to the javadocs.